scripts/lib/CIME/hist_utils.py

```diff
@@ -183,7 +145,8 @@ def _hists_match(model, hists1, hists2, suffix1="", suffix2=""):
 
     for hists, suffix, normalized, multi_normalized in [(hists1, suffix1, normalized1, multi_normalized1), (hists2, suffix2, normalized2, multi_normalized2)]:
         for hist in hists:
-            normalized_name = hist[hist.rfind(model):]
+            offset = max(0,hist.rfind(model))
```

Comment:
I have generalized so that history files do not need to include the model name.  I'm sure it's not 100% fool proof.  If the normalized filename does something like leave the date out then either we will need to improve the scripts or the user / component developer could consider improving their file naming convention.  @mvertens do we currently have documented the history file naming convention for cime?

Reply:
Here are the current file name conventions for CESM: http://www.cesm.ucar.edu/models/cesm2/naming_conventions.html#outputFilenames

I feel fairly strongly that, if we are going to generalize these, then we need to clearly document what we do (and possibly don't) expect to work. Patterns that are expected to work should be verified via unit tests of the relevant functions. Saying something to the effect of, "We kind of support any file names, but some things may not work if you depart too far from our existing files" (I'm paraphrasing here: I know you didn't say that directly) is confusing for component model developers and makes it very hard for cime developers to know what they are supposed to support in the scripts. While the ideal case would be to have a less restrictive file naming convention that is both clearly documented and well tested, I'd rather have something restrictive but clearly documented than something less restrictive but vague and uncertain.

I sense that an in-person discussion may be needed here. We could discuss this at the cime telecon Jan 8, but it might also be good if you and I talk along with @mvertens this week.